src/main/java/net/dries007/tfc/types/DefaultRecipes.java

```diff
@@ -134,10 +138,168 @@ public static void onRegisterBarrelRecipeEvent(RegistryEvent.Register<BarrelReci
             // this ratio works for 9b + 1b = 10b (full barrel) of brine/milk_vinegar, but leaves odd ninths of fluid around for other mixtures.
             //  previous todo had "make it a simpler calculation"
             new BarrelRecipe(IIngredient.of(FRESH_WATER.get(), 500), IIngredient.of("dustFlux"), new FluidStack(LIMEWATER.get(), 500), ItemStack.EMPTY, 0).setRegistryName("limewater"),
+            new BarrelRecipe(IIngredient.of(LIMEWATER.get(), 100), IIngredient.of(ItemsTFC.GYPSUM), null, new ItemStack(BlocksTFC.ALABASTER_RAW_PLAIN), ICalendar.TICKS_IN_HOUR).setRegistryName("plain_alabaster"),
 
             new BarrelRecipeTemperature(IIngredient.of(FRESH_WATER.get(), 1), 50).setRegistryName("fresh_water_cooling"),
             new BarrelRecipeTemperature(IIngredient.of(SALT_WATER.get(), 1), 50).setRegistryName("salt_water_cooling")
         );
+
+        //Nice handy list of dye in the correct order BY META DATA to get the correct fluid.
+        List<FluidWrapper> fluids = new ArrayList<>();
+        fluids.add(WHITE_DYE);
+        fluids.add(ORANGE_DYE);
+        fluids.add(MAGENTA_DYE);
+        fluids.add(LIGHT_BLUE_DYE);
+        fluids.add(YELLOW_DYE);
+        fluids.add(LIME_DYE);
+        fluids.add(PINK_DYE);
+        fluids.add(GRAY_DYE);
+        fluids.add(LIGHT_GRAY_DYE);
+        fluids.add(CYAN_DYE);
+        fluids.add(PURPLE_DYE);
+        fluids.add(BLUE_DYE);
+        fluids.add(BROWN_DYE);
+        fluids.add(GREEN_DYE);
+        fluids.add(RED_DYE);
+        fluids.add(BLACK_DYE);
+
+        //The many many many recipes that is dye. This assumes that the standard meta values for colored objects are followed.
+        for (EnumDyeColor dyeColor : EnumDyeColor.values())
+        {
+            String dyeName = dyeColor.getDyeColorName();
```

Comment:
`EnumDyeColor#getDyeColorName` is client only, don't use.
```suggestion
            String dyeName = dyeColor.getName();
            if (dyeColor == EnumDyeColor.SILVER)
            {
                dyeName = "light_gray";
            }
```